scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -489,6 +496,9 @@ def _parse_report_file(self, report):
                         tc.status = 'error'
                     tc.reason = elem.get('message')
                     tc.output = elem.text
+        else:
+            self.state = 'skipped'
+            self.instance.reason = 'No tests collected'
```

Comment:
This is sth extra than described in the commit. Can you make an extra commit out of it for the sake of bisectability?

Reply:
This was added just to not fail tests when all testcases were filtered. Just you will find in the output that test is skipped, because of 'No test collected` from the pytest execution:
```
./zephyr/scripts/twister -vv -T zephyr/samples/subsys/testsuite/pytest/shell -p native_sim --pytest-args='-k test_not_exists'
...
DEBUG   - PYTEST: ============================ no tests ran in 0.00s =============================
DEBUG   - run status: native_sim/samples/subsys/testsuite/pytest/shell/sample.pytest.shell skipped
INFO    - 1/1 native_sim                samples/subsys/testsuite/pytest/shell/sample.pytest.shell SKIPPED (No tests collected)
INFO    - 1 test scenarios (1 test instances) selected, 1 configurations skipped (0 by static filter, 1 at runtime).
```
Should it really be a change in a separate commit? IMO it is small change that is in line with other changes in that PR